setup.py

```diff
@@ -35,7 +35,7 @@
         "nbformat",
         "pyyaml",
         "toml",
-        "markdown-it-py>=1.0.0,<3.0.0",
+        "markdown-it-py>=1.0.0,<4.0",
```

Comment:
Let's get rid of the upper bound. You can add the future release in the experimental CI if you're afraid of major number changing an api you are using. If you think the api you use is stable, I would suggest to get rid of the upper bound altogether. Relevant errors will show up in the CI or a simple scheduled checker

Reply:
Also in the CI, you should use `~=` to enforce a specific version, otherwise other dependencies might pul your dependency down and you are not properly checking the version you expect. It is a pain to maintain it though, so I would suggest to only do it for the hard/core dependencies (I am not sure if this project has any of them. Stuff like `pyyaml` are too big and can rely on their stability)